packages/console/src/hooks/use-ui-languages.ts

```diff
@@ -0,0 +1,39 @@
+import { builtInLanguages as builtInUiLanguages } from '@logto/phrases-ui';
+import { useMemo } from 'react';
+import useSWR from 'swr';
+
+import { CustomPhraseResponse } from '@/types/custom-phrase';
+
+import { RequestError } from './use-api';
+
+const useUiLanguages = () => {
+  const {
+    data: customPhraseList,
+    error,
+    mutate,
+  } = useSWR<CustomPhraseResponse[], RequestError>('/api/custom-phrases');
+
+  const isLoading = useMemo(() => !customPhraseList && !error, [customPhraseList, error]);
+
+  const languages = useMemo(
+    () =>
+      [
+        ...new Set([
+          ...builtInUiLanguages,
+          ...(customPhraseList?.map(({ languageTag }) => languageTag) ?? []),
+        ]),
+      ]
+        .slice()
+        .sort(),
+    [customPhraseList]
+  );
+
+  return {
```

Comment:
nit: we can return data in the form of "SWR style":
```ts
return { data, error, mutate }
```

Reply:
Just keep aligned to `useSettings` and other swr wrappers.

Reply:
Can you add an issue to refactor them?